File: ZooZExtendedServerAPI-Java/src/main/java/com/zooz/extended/java/lib/ZooZQueryTypes/CustomerLoginNameQuery.java
package com.zooz.extended.java.lib.ZooZQueryTypes;

import com.zooz.extended.java.lib.model.NVPs;
import com.zooz.extended.java.lib.model.QueryType;
import com.zooz.extended.java.lib.model.enums.Queries;

/**
 * Get all the transactions details followed by a specific customer login name
 */
public class CustomerLoginNameQuery extends QueryType {

    private String customerLoginName;

    /**
     * @param customerLoginName: The customer login name on the merchant's app
     */
    public CustomerLoginNameQuery(String customerLoginName) {
        this.customerLoginName = customerLoginName;
    }

    @Override
    protected void addQueryParametersToNVP(NVPs nvps) {
        nvps.add("customerLoginName", customerLoginName);
    }

    @Override
    protected Queries getQuery() {
        return Queries.customerLoginName;
    }
}

File: PHP/ZooZQueries/CustomerLoginNameQuery.php
<?php

require_once 'lib/QueryType.php';
require_once './AbstractClasses/EnumQueries.php';

/**
 * Get all the transactions details followed by a specific customer login name
 */
class CustomerLoginNameQuery extends QueryType{
    
    private $customerLoginName;
    /**
     * 
     * @param type $customerLoginName - The customer login name on 
     *  the merchant's app
     */
    function __construct($customerLoginName) {
        parent::__construct();
        $this->customerLoginName = $customerLoginName;
    }
    
    protected function addQueryParametersToTheList($nvps) {
        $nvps->add("customerLoginName", $this->customerLoginName);
    }

    protected function getQueryType() {
        return EnumQueries::customerLoginName;
    }
}
